### ISSUE TITLE
Issue with dataset names on SDE databases Raster/Feature

### ISSUE THREAD

**Issue by TomWaterat** [NSW-OEH-EMS-KST/grid-garage-3#224]
When the input for any of the tools is a raster on an SDE database the output files are all assigned the same name of "Grid_SDE".

**Comment by byezy** [comment]
The name derives from the output of arcpy.describe(dataset). So do a describe on these rasters and see if the proper names are in any of the output fields... this is more likely a bug in the esri function...

**Comment by TomWaterat** [comment]
Thanks Dave, See screen grab of Describe output. Could it be the second "." in the name? i.e. "Grid.SDE.ASDST_ACM_IMP_V7" is truncated to "Grid_SDE".
![describe_grid_gdb](https://user-images.githubusercontent.com/4453449/42742608-8ae8f79e-88ab-11e8-8c52-32c05678e842.JPG)

**Comment by byezy** [comment]
Yep its our function to sanitise the raster names for the various workspaces. Wonder why underscores were used until this unnecassary name prefix grid.sde.  amatuers at work again?  Are you able to catalog the actual workspace as it looks like you are describing connections not datasets... its just that the connection has some dataset-like properties... if you can search for data in the actual workspace you might get further...

**Comment by byezy** [comment]
And on further thought... im surprised that the tools would take a connection file as a dataset input... must be some esri magic!

**Comment by byezy** [comment]
And you cant assign me to squat bro ;-) LOL gizza buzz at work tomorrow we can troubleshoot it a bit. First thing search for rasters in that sde workspace...

**Comment by byezy** [comment]
This is an issue for the minute, I can't find a single raster housed on any SDE available to me sorry mate. How is that citrix/remote thing looking...?

**Comment by byezy** [comment]
Hi Tom, could you knock up a little test for this sometime, the screen shot you posted above actually looks right to me at the moment, obviously I am still pulling on my GG beanie snugly...

**Comment by TomWaterat** [comment]
Hi Dave, I'm not sure how to create an SDE from scratch.. But from memory the GG Search tool wasn't returning the complete gis file name for any files. I'll try and replicate.

**Comment by byezy** [comment]
Hi Tom, after we have bedded down the first set of changes for describing (i219) and after a pull ;-) can you please run this scenario again with the **new geodata & raster describe** tools.

**Comment by byezy** [comment]
With output from the new tools as requested above, I will try to mock up a test to fix this without access to actual SDE data...

**Comment by byezy** [comment]
This is the same issue that arises with say, the feature-copy tool. test data needed.

**Comment by byezy** [comment]
@TomWaterat I have an idea bro. I will make a test branch for this issue... could you push in a table with some rasters and vectors out of an SDE please? I know you need this to work, I will get it going for ya. But some real names to work with would be best which I don't have...

**Comment by byezy** [comment]
@TomWaterat I have made branches **i224** 

**NB in the test repo, if you push anything in please do so in a new folder within the branch, as if I have stuff checked out it is not possible to resolve diffs in the binaries for, say, and fgdb or mxd...**

**Comment by TomWaterat** [comment]
I've created a test branch (i224a). I ran the 'Converstion Tools > To Geodatabase > Feature Class to Geodatabase' tool to copy a bunch of features from an SDE to a file geodatabase. ArcGIS strips out the first two parts of the name 'SoilDB' and 'SDE' uses the last part of the name - the basin name in this case. I also generated a bunch of grids with full-stops in the name.. Your code converts them to underscores which is fine.. but this does not happen when the input is an SDE name.
![Bug_SDE2](https://user-images.githubusercontent.com/4453449/61613266-38dafb00-aca4-11e9-865b-039de0245708.JPG)
